saleor/graphql/product/mutations/products.py

```diff
@@ -846,3 +851,161 @@ def mutate(cls, root, info, image_id, variant_id):
+    content = graphene.Field(DigitalContent)
+
+    class Arguments:
+        id = graphene.ID(
+            description='ID of a product variant to upload digital content.',
+            required=True)
+        input = ProductVariantDigitalUploadInput(
+            required=True,
+            description='Fields required to create a digital content.')
+
+    class Meta:
+        description = dedent('''Create new digital content. This mutation must 
+        be sent as a `multipart` request. More detailed specs of the upload 
+        format can be found here:
+        https://github.com/jaydenseric/graphql-multipart-request-spec''')
+
+    @classmethod
+    @permission_required('product.manage_products')
+    def mutate(cls, root, info, id, input):
+        errors = []
+        variant = cls.get_node_or_error(
+            info, id, errors, 'id', only_type=ProductVariant)
+        content_data = info.context.FILES.get(input['content_file'])
+        if hasattr(variant, 'digital_content'):
+            msg = 'Variant %s already has digital content' % id
+            cls.add_error(errors, 'id', msg)
+
+        digital_content = None
+        if not errors:
+            digital_content = models.DigitalContent(
+                content_file=content_data
+            )
+            default_max_download = get_default_digital_max_downloads()
+            max_downloads = input.get('max_downloads', default_max_download)
+            if max_downloads:
+                digital_content.max_downloads = max_downloads
+
+            default_url_valid_days = get_default_digital_url_valid_days()
+            url_valid_days = input.get('url_valid_days', default_url_valid_days)
+            if url_valid_days:
+                digital_content.url_valid_days = url_valid_days
+
+            default_automatic_fulfillment = get_default_automatic_fulfillment()
+            automatic_fulfillment = input.get(
+                'automatic_fulfillment', default_automatic_fulfillment)
+            if automatic_fulfillment:
+                digital_content.automatic_fulfillment = automatic_fulfillment
+
+            variant.digital_content = digital_content
+            variant.digital_content.save()
+        return ProductVariantDigitalUpload(
+            content=digital_content, errors=errors)
+
+
+class ProductVariantDigitalDelete(BaseMutation):
+    variant = graphene.Field(ProductVariant)
+
+    class Arguments:
+        id = graphene.ID(
+            description='ID of a product variant with digital content to remove.',
+            required=True)
+
+    class Meta:
+        description = dedent(
+            'Remove digital content assigned to given variant')
+
+    @classmethod
+    @permission_required('product.manage_products')
+    def mutate(cls, root, info, id):
+        errors = []
+        variant = cls.get_node_or_error(
+            info, id, errors, 'id', only_type=ProductVariant)
+
+        if not hasattr(variant, 'digital_content'):
```

Comment:
I think here as well we can simply do nothing if there is no content. Mutation result will anyway return a variant without any digital content.